packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3279,32 +3278,48 @@ export class ContainerRuntime
 		address: string | undefined,
 		type: string,
 		content: any,
+		targetClientId?: string,
 	): ISignalEnvelope {
-		const newSequenceNumber = ++this._signalTracking.signalSequenceNumber;
-
-		// Initialize tracking to expect the first signal sent by the connected client.
-		if (
-			this._signalTracking.minimumTrackingSignalSequenceNumber === undefined ||
-			this._signalTracking.trackingSignalSequenceNumber === undefined
-		) {
-			this._signalTracking.minimumTrackingSignalSequenceNumber = newSequenceNumber;
-			this._signalTracking.trackingSignalSequenceNumber = newSequenceNumber;
-			this._signalTracking.baseSignalTrackingGroupSequenceNumber = newSequenceNumber;
+		const isExpectedToReturn =
+			targetClientId === undefined || targetClientId === this.clientId;
+
+		// clientsignalsSequenceNumber is solely used for local signal tracking/telemetry purposes,
+		// so therefore we will only increment CSSN when the signal is expected to return.
+		// Targeted signals to remote clients will be ignored as they are not expected to come back.
+		const newSequenceNumber = isExpectedToReturn
+			? ++this._signalTracking.signalSequenceNumber
+			: -1;
```

Comment:
We only are using this for self signal tracking, but this feels wrong.
If that field really only matters for self signal tracking then let's update the name and make it optional. I would go ahead as remove the self-targeted client possibility too. Just do it for broadcasts.
(If we do general numbering for signals, I think is should go into ISignalMessageBase clientConnectionNumber at a later date.)